### DISCUSSION TITLE
Avian ESC support

### DISCUSSION THREAD

**Discussion by SCMicroMike** [iNavFlight/inav#10255] [General]
I have been looking to d an eflite Viper 90 with iNav. It comes with a 130A Avian smart ESC. The protocol seems to be smart throttle. This enables throttle control, throttle reverse and telemetry. Is this ESC protocol available to be supported in iNav?

**Comment by RobertoD91** [comment]
I was about to ask the same question.
I also have an Avian ESC and would like to reverse the motor from the radio.
~Does anyone know if the ESC uses the SRXL2 protocol?~
EDIT: confirmed SRXL2
If it does, that’s great news because it’s open source ( https://github.com/SpektrumRC/SRXL2/ ) and already integrated into iNav for the receiver.

**Comment by UltraFly** [comment]
I'm looking in to implementing this and testing on a radiomaster nexus xr. i have created a inav branch to work on it. I'll post here on my progress

**Reply by RobertoD91** [reply]
> I'm looking in to implementing this and testing on a radiomaster nexus xr. i have created a inav branch to work on it. I'll post here on my progress

Great! I have a working PoC ( Claude genereted.. srxl2 handshake works ) in my repo, give a look:
- https://github.com/RobertoD91/inav/tree/claude/esc-spektrum-protocol-check-i5b6jx  
- https://github.com/RobertoD91/inav-configurator/tree/claude/esc-spektrum-protocol-check-i5b6jx

**Comment by RobertoD91** [comment]
Related issue: https://github.com/iNavFlight/inav/issues/11184
